### PR TITLE
feat: added liked, history and watch later functionalities

### PR DESCRIPTION
**Pull request by md-amir-gauhar**

### CODE REVIEW

File: src/components/VideoCard.jsx [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
 const VideoCard = ({ _id, title, createdAt, views, likes, creator, creatorImg }) => {
   const navigate = useNavigate()
-  const onClickHandler = () => {
+
+  const onClickHandler = (_id) => {
     navigate(`/video/${_id}`)
+    try {
+      (async () => {
+        const data = {
+          // _id,
```
Comment (Pratik1005) [inline src/components/VideoCard.jsx:17]: remove commented code if not needed

File: src/components/VideoCard.jsx [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
+          id: _id,
+          title,
+          creator,
+        }
+        const response = await axios.post("/api/user/history", { video: data }, {
+          headers: {
+            authorization: getAuthData()
+          }
+        })
+        console.log(response);
```
Comment (Pratik1005) [inline src/components/VideoCard.jsx:27]: remove log

File: src/pages/History.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  useEffect(() => {
+    try {
+      (async () => {
+        const res = await axios.get('/api/user/history', {
+          headers: {
+            authorization: getAuthData()
+          }
+        })
+
+        console.log(res.data.history)
```
Comment (Pratik1005) [inline src/pages/History.js:63]: same here

File: src/pages/VideoPage.js [hunk 4]
```diff
@@ -1,31 +1,174 @@
...
+    }
+
+    try {
+      (async () => {
+        const response = await axios.post("/api/user/history", { video: data }, {
+          headers: {
+            authorization: getAuthData()
+          }
+        })
+        console.log(response);
```
Comment (Pratik1005) [inline src/pages/VideoPage.js:48]: same here

**Review by Pratik1005 — APPROVED** [review]
Rest looks good

**Review by Hamza-777 — APPROVED** [review]
Looks good to merge

**Pull request merged**